hail/src/main/scala/is/hail/rvd/RVDType.scala

```diff
@@ -49,6 +49,48 @@ final case class RVDType(rowType: TStruct, key: IndexedSeq[String] = FastIndexed
       other.kFieldIdx,
       false)
 
+  def intervalJoinComp(other: RVDType): UnsafeOrdering = {
```

Comment:
this looks right, but the logic is sufficiently complicated that I'd prefer you add a comment on the conceptual rules for comparing an interval and a point (including the half-defined case).